homeassistant/components/apcupsd/binary_sensor.py

```diff
@@ -62,14 +62,23 @@ def __init__(
             )
         self.entity_description = description
         self._data_service = data_service
+        self._attr_available = False
 
     def update(self) -> None:
         """Get the status report from APCUPSd and set this entity's state."""
-        self._data_service.update()
+        try:
+            self._data_service.update()
+            self._attr_available = True
 
-        key = self.entity_description.key.upper()
-        if key not in self._data_service.status:
-            self._attr_is_on = None
-            return
+            key = self.entity_description.key.upper()
+            if key not in self._data_service.status:
+                self._attr_is_on = None
+                return
 
-        self._attr_is_on = int(self._data_service.status[key], 16) & VALUE_ONLINE > 0
+            self._attr_is_on = (
+                int(self._data_service.status[key], 16) & VALUE_ONLINE > 0
+            )
```

Comment:
Only `update()` is able to throw the `OSError` in this case, and thus should be the only part that is wrapped in the try-block.

Reply:
Thanks!
Should be fixed now.